Source/Methods/Envelope Tracking/EnvelopeTracking.cs

```diff
@@ -0,0 +1,309 @@
+            {
+                return new TrackerConfiguration()
+                {
+                    InputImpedance_Ohms = 1e6,
+                    CommonModeOffset_V = 1.0,
+                    Gain_VperV = 2.5,
+                    OutputOffset_V = 2.55
+                };
+            }
+        }
+
+        public enum DetroughType { 
+            Exponential, 
+            Cosine, 
+            Power
+        };
+
+        public struct DetroughConfiguration
+        {
+            public DetroughType Type;
+            public double MinimumVoltage;
+            public double MaximumVoltage;
+            public double Exponent;
+
+            public static DetroughConfiguration GetDefault()
+            {
+                return new DetroughConfiguration()
+                {
+                    Type = DetroughType.Exponential,
+                    MinimumVoltage = 1.5,
+                    MaximumVoltage = 3.5,
+                    Exponent = 1.2
+                };
+            }
+        }
+
+        public struct LookUpTable
+        {
+            public float[] DutInputPower_dBm;
+            public float[] SupplyVoltage_V;
+        }
+
+        public struct SynchronizationConfiguration
+        {
+            public double RFDelayRange_s;
+            public double RFDelay_s;
+
+            public static SynchronizationConfiguration GetDefault()
+            {
+                return new SynchronizationConfiguration()
+                {
+                    RFDelayRange_s = 2e-6, // +/- 1us
+                    RFDelay_s = 0.0
+                };
+            }
+        }
+        #endregion
+
+        #region Envelope Creation
+        public static Waveform CreateEnvelopeWaveform(Waveform referenceWaveform, DetroughConfiguration detroughConfig)
+        {
+            Waveform envelopeWaveform = new Waveform()
+            {
+                Name = referenceWaveform.Name + "Envelope",
+                Data = referenceWaveform.Data.Clone(),
+                SignalBandwidth_Hz = 0.8 * referenceWaveform.SampleRate,
+                PAPR_dB = double.NaN, // unnecessary to calculate as it won't be used
+                BurstLength_s = double.NaN, // not applicable in ET
+                SampleRate = referenceWaveform.SampleRate,
+                // burst start and stop locations will be left as null since they also will not be used
+                IdleDurationPresent = referenceWaveform.IdleDurationPresent, // can't null a bool, so copy from reference waveform
+                RuntimeScaling = 10 * Math.Log10(0.9), // applies 10% headroom to the waveform at runtime
+                Script = referenceWaveform.Script // we will copy the script but call replace on it in the following lines to change the waveform name
+            };
+            envelopeWaveform.Script = envelopeWaveform.Script.Replace(referenceWaveform.Name, envelopeWaveform.Name); // this is better here since now we only have to add the suffix once

```

Comment:
Once #103 is merged, switch this call to use my new `Waveform.UpdateWaveformNameAndScript`

Reply:
a8f8cda235e8a6f1ae0f38394088c1bcf228ae06